pkgs/development/tools/pnpm/fetch-deps/default.nix

```diff
@@ -14,6 +14,8 @@
     {
       hash ? "",
       pname,
+      pnpmWorkspace ? "",
+      prePnpmInstall ? "",
```

Comment:
> i think it would be much better to just use a built in phase here
> 
> i.e., if someone wants to set specific options or run arbitrary commands, they can just use `postConfigure`/`postInstall`/etc from `mkDerivation` rather than this (single) unique phase
> 
> this would require splitting up some of the internal phases here, namely when `HOME`, `store-dir`, and other defaults are applied - they should probably in the `configurePhase`

Hmm you have a point, but still though, even if I'd simply move the `runHook preInstall` line after `HOME` is set, which would allow to perform such an override, I'd have to do something like this:

```nix
stdenv.mkDerivation (finalAttrs: {
  # ...
  pnpmDeps = (pnpm_8.fetchDeps {
    inherit (finalAttrs) pname version src pnpmWorkspace;
    hash = "sha256-n7HTd/rKxJdQKnty5TeOcyvBU9j/EClQ9IHqbBaEwQE=";
  }).overrideAttrs(_: {
    preInstall = finalAttrs.prePnpmInstall;
  });

  prePnpmInstall = ''
    pnpm config set dedupe-peer-dependents false
  '';
  # ...
})
```

Seems much more ugly. Even if we'll just rename the argument to `preInstall`, such that the interface would look like this:

```nix
stdenv.mkDerivation (finalAttrs: {
  # ...
  pnpmDeps = pnpm_8.fetchDeps {
    inherit (finalAttrs) pname version src pnpmWorkspace;
    preInstall = finalAttrs.prePnpmInstall;
    hash = "sha256-n7HTd/rKxJdQKnty5TeOcyvBU9j/EClQ9IHqbBaEwQE=";
  };

  prePnpmInstall = ''
    pnpm config set dedupe-peer-dependents false
  '';
  # ...
})
```

It is still a bit weird IMO how these arguments are named differently, since they serve exactly the same purpose.

If you weren't thinking of either of these interfaces, I didn't fully understand you.